ArrayGenerica.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+template <class T>
+class ArrayGenerica {
+private:
+    T** elementos;
+    int capacidad;
+    int cantidad;
+
+public:
+    ArrayGenerica(int max) {
+        elementos = new T * [max];
+        capacidad = max;
+        cantidad = 0;
+    }
+
+    void agregar(T* elemento) {
+        if (cantidad < capacidad) {
+            elementos[cantidad++] = elemento;
+        }
+        else {
+            //throw CapacidadSuperadaException(cantidad, capacidad);
+        }
+    }
+    
+    void mostrar() {
+        for (int i = 0;i < cantidad;i++) {
+            elementos[i]->mostrar();
+        }
+    }
+    /*------------------------------------------------------------------------------*/
+
+    float promedio(float (*f)(T*)) {
+        float suma = 0;
+        for (int i = 0; i < cantidad; i++) {
+            suma += (*f)(elementos[i]);
+        }
+        return suma / cantidad;
+    }

```

Comment:
Nombre descriptivos y falto aplicar la función
```suggestion
    float getPromedio(float (*getAttr)(T*)) {
        float suma = 0;
        for (int i = 0; i < cantidad; i++) {
            suma += getAttr( elementos[i] )
        }
        return suma / cantidad;
    }

```